src/com/docker/projects/Documentation.groovy

```diff
@@ -0,0 +1,51 @@
+
+
+package com.docker.projects;
+
+// Add the following code to any Jenkinsfile.
+// It will return with success if there are no changes in the specified docs dir
+// and Will return with success if run on a non Pull Request branch
+
+// def docs = new com.docker.projects.Documentation()
+// docs.documentationPRChecker("docs")
+
+// docsDir ~~ the subdirectory within your repo that the documentation resides
+def documentationPRChecker(docsDir) {
+  if (env.CHANGE_ID != null) {
+    stage "docs PR checker"
+
+    tokens = "${env.JOB_NAME}".tokenize('/')
+    org = tokens[0]
+    repo = tokens[1]
+    branch = tokens[2]
+    sh "git log --format='%H' -1 > tempoutput"
+    sha = readFile("tempoutput").trim()
+    imageName = "${repo}:${branch}"
+    containerName = "${repo}-${branch}-${env.BUILD_ID}"
+
+    sh "env > env.txt" // in case I need to debug
+
+    sh "git log origin/master..${sha} ${docsDir} > tempoutput"
+    changes = readFile("tempoutput").trim()
+    if (changes.size() == 0) {
+      echo "no changes found in ${docsDir}"
```

Comment:
yeah, I grew up in a time where having multiple `return`'s was considered bad style - I'm not fully go-styled yet :)